File: Modules/Common/Resources/SIE.Web.Resources/WipResources/Commands/SynWipResSettingCommand.js
SIE.defineCommand('SIE.Web.Resources.WipResources.Commands.SynWipResSettingCommand', {
    meta: { text: "资源同步设置", group: "business", iconCls: "icon-ListConfig icon-blue" },
    execute: function (view, source) {
        var me = this;
        var commit = function (settings, win) {
            var me = this;
            var datas = [];
            settings.items.forEach(function (item) { datas.push(item.data); });
            view.execute({
                data: datas,
                success: function (res) {
                    var errMsg = res.Result;
                    if (errMsg == true) {
                        win.close();
                    }
                }
            });
        };
        me.showSetingDialog(commit);
    },
    showSetingDialog: function (commit) {
        SIE.AutoUI.getMeta({
            model: 'SIE.Resources.WipResources.SynWipResSetting',
            ignoreCommands: true,
            isDetail: false,
            ignoreQuery: true,
            callback: function (res) {
                var listView = SIE.AutoUI.createListView(res);
                var filter = {
                    Method: 'GetSynWipResSettings',
                    Parameters: []
                };
                var win = SIE.Window.show({
                    title: '资源同步设置'.L10N(),
                    width: 800,
                    height: 450,
                    items: listView.getControl(),
                    callback: function (btn) {
                        if (btn === "确定".t()) {
                            commit(listView.getData().data, win);
                            view.loadData();
                            return false;
                        }
                    }
                });
                filter = Ext.encode(filter);
                listView.loadData({
                    filter: filter,
                    action: 'queryer',
                    type: 'SIE.Web.Resources.WipResources.WipResourceDataQueryer'
                });
            }
        });
    }
});

File: Modules/Common/Resources/SIE.Web.Resources/WipResources/Commands/SynWipResSettingCommand.cs
using SIE.Common;
using SIE.Domain;
using SIE.Resources.WipResources;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.Resources.WipResources.Commands
{
    /// <summary>
    /// 资源同步设置命令
    /// </summary>
    [JsCommand("SIE.Web.Resources.WipResources.Commands.SynWipResSettingCommand")]
    public class SynWipResSettingCommand : ViewCommand
    {
        /// <summary>
        /// 命令执行逻辑
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>成功</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var settings = args.Data.ToJsonObject<List<SynWipResSetting>>();
            if (settings == null)
                throw new ArgumentNullException("同步设置数据为空".L10N());
            if (settings.Count > 0)
            {
                RF.Save(settings.AsEntityList());
                RT.Service.Resolve<WipResourceController>().RunSync();
            }
            return true;
        }
    }
}